### PR TITLE
Ccip 1039 publish test image on each release with release tag

### PR DESCRIPTION
**Pull request by jasonmci**
## Solution

### CODE REVIEW

File: .github/workflows/integration-tests-publish.yml [hunk 1]
```diff
@@ -19,6 +21,9 @@ jobs:
     name: Publish Integration Test Image
     runs-on: ubuntu-latest
     steps:
+      - name: Extract tag name
+        id: extract_tag
+        run: echo "::set-output name=tag::$(git describe --tags --match 'v*' --exact-match 2>/dev/null)"
```
Comment (chainchad) [inline .github/workflows/integration-tests-publish.yml]: You might be able to get rid of this step completely. If you do need output, I would go with the new format (set-output is [deprecated](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)): `echo "tag=$(...)" >> "$GITHUB_OUTPUT"`

Comment (chainchad) [reply .github/workflows/integration-tests-publish.yml]: This workflow runs on other triggers besides a tag push. When it runs on a branch, should you completely avoid setting a `RELEASE_TAG`?

File: .github/workflows/integration-tests-publish.yml
Comment (chainchad) [inline .github/workflows/integration-tests-publish.yml:1]: We should be able to get rid of this:
```

env:
  ECR_TAG: ${{ secrets.QA_AWS_ACCOUNT_NUMBER }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/chainlink-ccip-tests:develop
```

Now that it's no longer ref'd, right?

File: .github/workflows/integration-tests-publish.yml [hunk 2]
```diff
@@ -31,10 +36,15 @@ jobs:
         uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
         with:
           ref: ${{ github.event.pull_request.head.sha }}
+      - name: Collect other tags
+        id: collect_other_tags
+        run: |
+          RELEASE_TAG="${{ steps.extract_tag.outputs.tag != '' && format('{0}.dkr.ecr.{1}.amazonaws.com/chainlink-ccip-tests:{2}', secrets.QA_AWS_ACCOUNT_NUMBER, secrets.QA_AWS_REGION, steps.extract_tag.outputs.tag) || '' }}"
```
Comment (chainchad) [inline .github/workflows/integration-tests-publish.yml]: We have to account for the fact that this workflow could run in situations where there's not a tag push (like branch or manual dispatch). For that, we usually add a conditional to the step:

`if: startsWith(github.ref, 'refs/tags/')`

Then, once you know it's a tag, you can refer to the tag name in the GitHub Actions context of `${{ github.ref_name }}` which should let you avoid the `extract_tag` tag step above.

If this is triggered on a branch push, is it OK if the input value for `other_tags` is an empty string? If so, I might do that.

**Review by chainchad — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-tests-publish.yml (hunk 1 above)
Comment (jasonmci) [reply .github/workflows/integration-tests-publish.yml]: resolved. removed this completely

File: .github/workflows/integration-tests-publish.yml
Comment (jasonmci) [reply .github/workflows/integration-tests-publish.yml:1]: I don't know how this is used. As long as I can have more than one tag on an image, I think it meets what people need. If we dont need it, I can remove it.

**Review by chainchad — APPROVED** [review]

**Review by AnieeG — APPROVED** [review]

**Review by connorwstein — APPROVED** [review]

**Pull request merged**